superset-frontend/src/features/alerts/components/NotificationMethod.tsx

```diff
@@ -133,25 +136,29 @@ export const NotificationMethod: FunctionComponent<NotificationMethodProps> = ({
               )}
               value={method}
             />
+            {method !== undefined && index !== 0 && !!onRemove ? (
```

Comment:
1. `index !== 0` why do we prevent user from deleting the first item?
2. I don't see "Add notification method" button anymore 🤔 

<img width="756" alt="image" src="https://github.com/apache/superset/assets/15073128/80444de7-435e-4d20-a09c-c68c16332441">

Reply:
@fisjac might want to look at this as well. This one was from [this ticket](https://app.shortcut.com/preset/story/77355/notification-method-bugs). I removed the option to add another method unless there is `> 1` method in `allowedNotificationMethods`

Reply:
I believe that this is the intended behavior to only show the Add notification button when additional notification options are available. Though I think we need to limit the add notification button to disappear when `NotificationSettings.length === notificationOptions.length`

Reply:
Yeah I think the button wasn't there because I didn't have Slack as an option 👍 so my bad here